elastic/kibana
# [EPM] Handle multi fields in index template generation

## Summary

Implements https://github.com/elastic/kibana/issues/60495

Adds handling of the `multi_fields` property for fields of type `text` and `keyword`. This mirrors the functionality in `libbeats`, see
* https://github.com/elastic/beats/blob/master/libbeat/template/processor.go#L222-L226
* https://github.com/elastic/beats/blob/master/libbeat/template/processor.go#L258-L262

The corresponding ES documentation does not restrict multi fields to `text` and `keyword` fields. We might at some point need/want to add multi fields to other field types too. 

There is some refactoring potential in `server/services/epm/elasticsearch/template/template.ts`: `generateMappings()` and `generateMultiFields()` perform a very similar task and could be rolled into one function that calls itself recursively. I intentionally left this for the future because I think it's very readable now, and not too complex.

## How to test this

* Inspect the unit tests, and compare with https://github.com/elastic/beats/blob/master/libbeat/template/processor_test.go#L159-L181
* Verify that package installation still works, and index templates are generated correctly.
